lib/stages/pulsarPostProcess.cpp

```diff
@@ -149,6 +149,7 @@ void pulsarPostProcess::main_thread() {
     }
     uint64_t frame_fpga_seq_num = 0;     // sample starting the current input frame
     uint32_t current_input_location = 0; // goes from 0 to _samples_per_data_set
+    ice_stream_id_t stream_id = ice_get_stream_id_t(in_buf[0], in_buffer_ID[0]);
```

Comment:
This needs to happen after line 193 the `sync_input_buffers()` call.  Before that call `in_buf[0]` might not have any frames with the metadata (because it hasn't made it to this part of the pipeline yet).

Reply:
thanks for catching that @andrerenard , made that change and pushed.